src/include/execution/sql/projected_row_wrapper.h

```diff
@@ -0,0 +1,65 @@
+#pragma once
+#include "execution/util/execution_common.h"
+#include "storage/projected_row.h"
+
+namespace terrier::execution::sql {
+
+/**
+ * Wrapper around projected rows.
+ * TODO(Amadou): This is not really necessary. The bytecode handlers can directly call the PR functions. But it's
+ * nice to have it as a separate class for now.
+ */
+class EXPORT ProjectedRowWrapper {
+ public:
+  /**
+   * Constructor
+   */
+  explicit ProjectedRowWrapper(storage::ProjectedRow *pr) : pr_{pr} {}
+
+  /**
+   * Get a pointer to the value in the column at index @em col_idx
+   * @tparam T The desired data type stored in the vector projection
+   * @tparam Nullable Whether the column is NULLable
+   * @param col_idx The index of the column to read from
+   * @param[out] null null Whether the given column is null
+   * @return The typed value at the current iterator position in the column
+   */
+  template <typename T, bool Nullable>
+  const T *Get(uint16_t col_idx, bool *null) const {
+    // NOLINTNEXTLINE: bugprone-suspicious-semicolon: seems like a false positive because of constexpr
+    if constexpr (Nullable) {
+      TERRIER_ASSERT(null != nullptr, "Missing output variable for NULL indicator");
+      *null = pr_->IsNull(col_idx);
+    }
+    return reinterpret_cast<T *>(pr_->AccessWithNullCheck(col_idx));
```

Comment:
Yes. There should be an `AccessWithoutNullCheck` call to speed up the case where we know it's not nullable.
Will fix it.

Reply:
Is @jrolli's comment still valid?

Reply:
The concern is still valid, but it is not a correctness issue.  We can leave a TODO for now, but this will likely need to be fixed in the long run since this will probably be on the critical path of every OLTP query.